readthedocs/organizations/forms.py

```diff
@@ -208,118 +210,77 @@ def __init__(self, *args, **kwargs):
+        # then try sending an invitation to it.
         try:
             validator = EmailValidator(code='lookup not an email')
-            validator(lookup)
-
-            member = (
-                User.objects.filter(
-                    emailaddress__verified=True,
-                    emailaddress__email=lookup,
-                    is_active=True,
-                ).first()
-            )
-            if member is not None:
-                return self.validate_member_user(member)
-
-            invite = TeamInvite(
-                organization=self.team.organization,
-                team=self.team,
-                email=lookup,
-            )
-
-            return self.validate_member_invite(invite)
+            validator(username)
+            return username
         except ValidationError as error:
             if error.code != 'lookup not an email':
                 raise
 
-        # Not an email, attempt username lookup
-        try:
-            member = User.objects.get(username=lookup, is_active=True)
-            return self.validate_member_user(member)
-        except User.DoesNotExist:
-            raise forms.ValidationError('User not found')
+        raise forms.ValidationError(
+            _("User %(username)s does not exist"), params={"username": username}
+        )
 
     def validate_member_user(self, member):
         """Verify duplicate team member doesn't already exists."""
         if TeamMember.objects.filter(team=self.team, member=member).exists():
             raise forms.ValidationError(_('User is already a team member'),)
         return member
 
-    def validate_member_invite(self, invite):
-        """
-        Verify team member and team invite don't already exist.
-
-        Query searches for duplicate :py:cls:`TeamMember` instances, and also
-        for existing :py:cls:`TeamInvite` instances, sharing the team and email
-        address of the given ``invite``
-
-        :param invite: :py:cls:`TeamInvite` instance
-        """
-        queryset = TeamMember.objects.filter(
-            Q(
-                team=self.team,
-                invite__team=self.team,
-                invite__email=invite.email,
-            ),
-        )
-        if queryset.exists():
-            raise forms.ValidationError(
-                _('An invitation was already sent to this email'),
+    def save(self):
+        """Create an invitation only if the user isn't already a member."""
+        user = self.cleaned_data["user"]
+        if isinstance(user, User):
+            # If the user is already a member or the organization
+            # don't create an invitation.
+            if (
+                AdminPermission.members(self.team.organization)
+                .filter(pk=user.pk)
+                .exists()
+            ):
+                member = self.team.organization.add_member(user, self.team)
+                member.send_add_notification(self.request)
+                return user
```

Comment:
I don't understand this. Why are we changing the flow here?

If I'm member of the organization and I'm invited to be part of another team, I'd expect to receive that invitation and accepted it as well. I think it's important to keep consistency here with the rest of the flow.

Reply:
If you are already a member of the organization, you don't need to confirm each time you are added to a team, only the first time you are invited to a new organization, otherwise it would be a pain for admins to add users to teams of the same organization, this is similar to github, you don't need to accept every time you are added to a team, only when added to an organization.